### PR TITLE
Make sure checkins from crons work and keep working

### LINKED ISSUE #3145: Test Celery Beat regression (2.4.0 through 2.5.0 inclusive)
We had a regression in our Celery Beat integration that prevented Cron checkins from being sent at the end of the cron job from version 2.4.0 through (including) 2.5.0.

We isolated the cause to c80cad1e6e17790f02b29115013014d3b4bebd3c, which we reverted in #3144. We released the fix in 2.5.1.

We should investigate the exact cause of this regression and add tests to ensure we cannot introduce a similar regression again.

### PR DESCRIPTION
**Pull request by antonpirker**
After refactoring the Celery Beat integration there was a regression, where the `ok`/`error` check ins for crons where not sent. 

We reverted the refactoring here: https://github.com/getsentry/sentry-python/pull/3144

This PR:
- Fixes the regression: 
  https://github.com/getsentry/sentry-python/pull/3155/files#diff-a05979007fa0398cc22f7b4910ccd3fde27fdea262a6137d1abdc676c633fcdcR75
- Adds tests that use a real Celery worker, real Celery beat, and real Redis:
  https://github.com/getsentry/sentry-python/pull/3155/files#diff-6080c59a09cd8a2064cb87c5e2512c0589dda02f0ad2f46c94f774b0e645676fR56

After we have merged this we can revert the revert linked above. 

Fixes #3145

### CODE REVIEW

File: tests/integrations/celery/test_celery.py [hunk 1]
```diff
@@ -26,9 +26,19 @@ def inner(signal, f):
 
 @pytest.fixture
 def init_celery(sentry_init, request):
-    def inner(propagate_traces=True, backend="always_eager", **kwargs):
+    def inner(
```
Comment (antonpirker) [inline tests/integrations/celery/test_celery.py:29]: I updated this `init_celery` to make sure we can also set `monitor_beat_tasks` here.

**Comment by antonpirker** [comment]
We have now the old `init_celery` fixture and a new `celery_init` fixture. 

The new one is way easier, and I think the old one is not needed anymore because advancements in Celery. So I think we can move to the new simpler fixture to be used everywhere.

File: sentry_sdk/integrations/celery/beat.py [hunk 2]
```diff
@@ -184,6 +189,11 @@ def _patch_redbeat_maybe_due():
 
     original_maybe_due = RedBeatScheduler.maybe_due
 
+    # Cant use __name__ here, because some of our tests mock original_apply_entry
```
Comment (sentrivana) [inline sentry_sdk/integrations/celery/beat.py:192]:
```suggestion
    # Cant use __name__ here, because some of our tests mock original_maybe_due
```

**Review by sentrivana — APPROVED** [review]
Looking good. In the longer term we should look into moving the old tests to the new setup too.

**Comment by antonpirker** [comment]
Closing in favor of https://github.com/getsentry/sentry-python/pull/3176

**Pull request closed without merging**